src/promptflow/promptflow/executor/_input_assignment_parser.py

```diff
@@ -49,13 +76,25 @@ def parse_node_property(node_name, node_val, property=""):
                 elif index.isdigit():
                     index = int(index)
                 else:
-                    raise InvalidReferenceProperty(f"Invalid index {index} when accessing property {property}")
+                    raise InvalidReferenceProperty(
+                        message_format=(
+                            "Invalid index '{index}' when accessing property '{property}' of the node '{node_name}'. "
```

Comment:
1. needs template
2. this is only when output property value is array ? maybe add this part into the message

Reply:
1. updated.
2. I tried building a test case for this error but failed.